sphinx/util/nodes.py

```diff
@@ -458,6 +467,8 @@ def make_refnode(builder, fromdocname, todocname, targetid, child, title=None):
 
 def set_source_info(directive, node):
     # type: (Any, nodes.Node) -> None
+    warnings.warn('set_source_info() is deprecated.',
+                  RemovedInSphinx40Warning)
```

Comment:
IMO, `set_source_info()` function will be replaced by `SphinxDirective.set_source_info()`. So I'd like to deprecate this in the future because it is no longer needed.
@shimizukawa what do you think?

Reply:
Finally, I reverted this.